### PR TITLE
URI Encode user input on zos-files commands

### LINKED ISSUE #1073: Downloading a file with "#" in member name failing
Trying to download all members in a file . But its failing as some of the members contain "#" in the member name .

**Command**  
zowe zos-files download all-members "DatasetName" -b -d "DirectoryLocation"

**Error Message :** 

z/OSMF REST API Error:
Rest API failure with HTTP(S) status 404
category: 1
rc:       4
reason:   3
message:  DatasetName(

Error Details:
HTTP(S) error status "404" received.
Review request details (resource, base path, credentials, payload) and ensure correctness.

Protocol:  https
Host:      host
Port:      port
Base Path: basepath
Resource:  /zosmf/restfiles/ds/DatasetName(member1)
Request:   GET
Headers:   [{"X-IBM-Data-Type":"binary"},{"Accept-Encoding":"gzip"},{"X-IBM-Response-Timeout":"250"},{"X-CSRF-ZOSMF-HEADER":true}]
Payload:   undefined

From User list thread 
https://lists.openmainframeproject.org/g/zowe-user/topic/84389452?p=Created,,,20,1,0,0::recentpostdate/sticky,,,20,2,0,84389452

Thanks,Ajit

### PR DESCRIPTION
**Pull request by awharn**
URI encodes all user input that goes into the URL for the zos-files APIs
Resolves #1073

### CODE REVIEW

**Review by KevinLoesch1 — APPROVED** [review]
LGTM

**Review by gejohnston — APPROVED** [review]

File: __tests__/__src__/TestUtils.ts
Comment (zFernand0) [inline __tests__/__src__/TestUtils.ts:1]: general question:
What's the difference between using these TestUtils vs using the `__tests__/__packages__/cli-test-utils/src/TestUtils.ts`?

Not asking to change anything on this PR, but should we eventually use the `@zowe/cli-test-utils`? 
(dogfooding 😋 )

File: packages/zosfiles/src/methods/delete/Delete.ts [hunk 1]
```diff
@@ -193,7 +191,7 @@ export class Delete {
         ImperativeExpect.toNotBeEqual(fileSystemName, "", ZosFilesMessages.missingFileSystemName.message);
 
         // Format the endpoint to send the request to
-        const endpoint = ZosFilesConstants.RESOURCE + ZosFilesConstants.RES_ZFS_FILES + "/" + fileSystemName;
+        const endpoint = ZosFilesConstants.RESOURCE + ZosFilesConstants.RES_ZFS_FILES + "/" + encodeURIComponent(fileSystemName);
```
Comment (zFernand0) [inline packages/zosfiles/src/methods/delete/Delete.ts:194]: curious as to why we don't use `posix.join` everywhere 😋

**Review by zFernand0 — APPROVED** [review]
LGTM! 😋

File: packages/zosfiles/src/methods/delete/Delete.ts (hunk 1 above)
Comment (awharn) [reply packages/zosfiles/src/methods/delete/Delete.ts:194]: Not sure, there are a ton of tests that will need to be updated in though. :)

File: packages/zosfiles/src/methods/copy/Copy.ts [hunk 2]
```diff
@@ -48,7 +48,7 @@ export class Copy {
         const endpoint: string = posix.join(
             ZosFilesConstants.RESOURCE,
             ZosFilesConstants.RES_DS_FILES,
-            toMemberName == null ? toDataSetName : `${toDataSetName}(${toMemberName})`
+            toMemberName == null ? encodeURIComponent(toDataSetName) : `${encodeURIComponent(toDataSetName)}(${encodeURIComponent(toMemberName)})`
```
[the remark this answers is not in the document]
Comment (awharn) [reply packages/zosfiles/src/methods/copy/Copy.ts:51]: That would behave mostly the same way, but the parenthesis would be encoded. Not sure how APIML or z/OSMF respond to that.

File: packages/zosfiles/CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,10 @@
 
 All notable changes to the Zowe z/OS files SDK package will be documented in this file.
 
+## Recent Changes
+
+- BugFix: Added URI encoding to user input that is sent to z/OSMF in the URL
```
[the remark this answers is not in the document]
Comment (awharn) [reply packages/zosfiles/CHANGELOG.md:7]: Yep!

File: packages/zosfiles/src/methods/copy/Copy.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (t1m0thyj) [reply packages/zosfiles/src/methods/copy/Copy.ts:51]: `encodeURIComponent` doesn't encode parentheses which is why I asked:
```
$ node -p "encodeURIComponent('abc(#def)')"
abc(%23def)
```

I'm fine with leaving this as-is though 🙂

**Review by t1m0thyj — APPROVED** [review]
LGTM, thanks @awharn!

File: packages/zosfiles/src/methods/copy/Copy.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (awharn) [reply packages/zosfiles/src/methods/copy/Copy.ts:51]: Ah, I'll update it then. :)

**Pull request merged**